hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -821,6 +865,38 @@ protected void finalizeWrite(HoodieTable<T, I, K, O> table, String instantTime,
     }
   }
 
+  private HoodieCommitMetadata resolveWriteConflictIfAny(HoodieTable<T, I, K, O> table, final String instantTime,
+                                            HoodieCommitMetadata thisCommitMetadata) {
+    Long currentTimeInSecs = System.currentTimeMillis() / 1000;
```

Comment:
I was expecting to use the atomic reference above. and just pull all the writes that happened after the `latestWriteInstantCompletedBeforeWriter` ? Also lets not assume that time in seconds is what we will use here.